src/main/scala/net/ruippeixotog/scalascraper/browser/HtmlUnitBrowser.scala

```diff
@@ -71,6 +73,25 @@ class HtmlUnitBrowser(browserType: BrowserVersion = BrowserVersion.CHROME) exten
     HtmlUnitDocument(window)
   }
 
+  def parseInputStream(inputStream: InputStream, charset: String): Document = {
+    val response = new WebResponse(getWebResponseData(inputStream, charset), getWebRequest(charset), 0)
+    val window = newWindow()
+    HTMLParser.parseHtml(response, window)
+    HtmlUnitDocument(window)
+  }
+
+  def getWebRequest(charset: String): WebRequest = {
```

Comment:
Can you make these two methods `private[this]` and move them to the end of the class, together with the other private methods?

Regarding this method specifically, there is already a `newRequest` here. Can you either merge them or give them more distinguishable names?